fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4

```diff
@@ -1941,13 +1941,18 @@ queryNoWith
     : queryPrimary (ORDER BY sortItem (',' sortItem)*)? (limitElement)?
     ;
 
-temporalClause
+queryPeriod
     : AS OF expression
     | FOR SYSTEM_TIME AS OF TIMESTAMP string
     | FOR SYSTEM_TIME BETWEEN expression AND expression
     | FOR SYSTEM_TIME FROM expression TO expression
     | FOR SYSTEM_TIME ALL
-    | FOR VERSION AS OF expression
+    | (FOR)? rangeType AS OF end=expression
+    ;
+
+rangeType
+    : TIMESTAMP
```

Comment:
We should not differ behaviors by TIMESTAMP and SYSTEM_TIME.

Reply:
There are a lot of historical reasons here. you could ref [patch1](https://github.com/apache/spark/commit/e99fdf9654481dd9b691a3c10e52f3f3db6ed2ba)  and [patch2](https://github.com/apache/spark/commit/f361ad8d2eb067ec86483a294771f1976f31a981).  If we compatible with the `SYSTEM_TIME`, we may also do this with `SYSTEM_VERSION` which is not supported now. The current syntax of starrocks is only used for mysql external table. and the added syntax of this patch designed to be compatible with most engine usages like spark or trino.  And one possible change is to merge `FOR SYSTEM_TIME AS OF TIMESTAMP string` into `(FOR)? rangeType AS OF end=expression`.  Do you have any suggestions?

Reply:
IMO the keyword `TIMESTAMP` is used to identify the string followed by is a TIMESTAMP type.
`(FOR)? periodType AS OF (TIMESTAMP)? end=expression`
For this syntax, I don't think this is right, because TIMESTAMP should belong to `expression`
It should be something, like `AS OF end = TIMESTAMP "2023-01-01 12:34:45"`

Reply:
https://en.wikipedia.org/wiki/SQL:2011
You can refer to the wiki to know more about it.

Reply:
And I think you can keep things simple by just using SYSTEM_TIME in starrocks, and you can achieve compatible with Trino in trino syntax. There is no need for us to introduce TIMESTAMP which is equal to SYSTEM_TIME here.

Reply:
done